dynamic_instrumentation/src/main.rs

```diff
@@ -61,6 +61,17 @@ struct Args {
     set_runtime: bool,
 
     /// `$RUSTFLAGS`, but as a CLI argument, not an inheritable environment variable.
+    /// These flags are appended (space-separated) to the existing `$RUSTFLAGS`, if it exists.
+    ///
+    /// The reason this exists is twofold:
+    ///
+    /// 1. It would be convenient if `cargo` itself had such a `--rustflags` argument,
+    ///    so at least we can recreate it here ourselves.
+    ///
+    /// 2. If this binary is invoked by something like `cargo run --bin c2rust-instrument`,
+    ///    then it's impossible to set `$RUSTFLAGS` only for the inner `cargo` that you want to add them to
+    ///    without also adding them to the outer `cargo run` `cargo`.
+    ///    `--rustflags` lets you do that easily.
```

Comment:
```suggestion
    /// Set the `RUSTFLAGS` environment variable for the instrumented Cargo invocation.
    ///
    /// This allows setting `RUSTFLAGS` for the inner `cargo` when c2rust-instrument is invoked via
    /// `cargo run`. If `RUSTFLAGS` is already set in the environment, these flags are appended to
    /// the old value following a space character.
```

Thanks for the more detailed documentation. Would you be opposed to something more concise, like the above?

Reply:
More concise, yes, but I'm not sure your suggestion captures the full intent.  Is there a reason the verbosity is bad?

```suggestion
    /// `$RUSTFLAGS`, but as a CLI argument, not an inheritable environment variable.
    /// These flags are appended (space-separated) to the existing `$RUSTFLAGS`, if it exists.
    ///
    /// The reason this exists is twofold:
    ///
    /// 1. It would be convenient if `cargo` itself had such a `--rustflags` argument,
    ///    so at least we can recreate it here ourselves.
    ///
    /// 2. If this binary is invoked by something like `cargo run --bin c2rust-instrument`,
    ///    then it's impossible to set `$RUSTFLAGS` only for the inner `cargo` that you want to add them to
    ///    without also adding them to the outer `cargo run` `cargo`.
    ///    `--rustflags` lets you do that easily.
```

Reply:
I prefer Frances wording as well, it communicates the information needed by users. We don't really need to document why the flag exists in the help text, that's more the place of comments in sources and commit messages. I appreciate the intent, and we should put this somewhere, I also just don't think it belongs here.

Reply:
It is also simultaneously the `rustdoc` documentation for that field, so I'm not sure where else to put it.

Reply:
How about this?

```rust
    /// Set `$RUSTFLAGS` for the instrumented `cargo`.
    ///
    /// This allows setting `$RUSTFLAGS` for the inner `cargo` when `c2rust-instrument` is invoked via `cargo run`, for example.
    /// If `$RUSTFLAGS` is already set, these `--rustflags` are appended with a space.
    //
    // The reason this exists is twofold:
    //
    // 1. It would be convenient if `cargo` itself had such a `--rustflags` argument,
    //    so at least we can recreate it here ourselves.
    //
    // 2. If this binary is invoked by something like `cargo run --bin c2rust-instrument`,
    //    then it's impossible to set `$RUSTFLAGS` only for the inner `cargo` that you want to add them to
    //    without also adding them to the outer `cargo run` `cargo`.
    //    `--rustflags` lets you do that easily.
```

Note that the bottom part use `//` not `///`.

Reply:
That works!

Reply:
Fixed in https://github.com/immunant/c2rust/pull/629/commits/7f608b9105b9978fac7a3f9c228b60ad6479da42.